pkg/token/interactive.go

```diff
@@ -51,32 +54,55 @@ func (p *InteractiveToken) Token() (adal.Token, error) {
 	clientOpts := azcore.ClientOptions{Cloud: cloud.Configuration{
 		ActiveDirectoryAuthorityHost: authorityFromConfig.String(),
 	}}
-	cred, err := azidentity.NewInteractiveBrowserCredential(&azidentity.InteractiveBrowserCredentialOptions{
-		ClientOptions: clientOpts,
-		TenantID:      p.tenantID,
-		ClientID:      p.clientID,
-	})
-	if err != nil {
-		return emptyToken, fmt.Errorf("unable to create credential. Received: %w", err)
-	}
+	scopes := []string{p.resourceID + "/.default"}
 
-	// Use the token provider to get a new token
-	interactiveToken, err := cred.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: []string{p.resourceID + "/.default"}})
-	if err != nil {
-		return emptyToken, fmt.Errorf("expected an empty error but received: %w", err)
-	}
-	if interactiveToken.Token == "" {
-		return emptyToken, errors.New("did not receive a token")
+	var token string
+	var expirationTimeUnix int64
+	var err error
+
+	if len(p.popClaims) > 0 {
+		// If PoP token support is enabled and the correct u-claim is provided, use the MSAL
+		// token provider to acquire a new token
+		token, expirationTimeUnix, err = pop.AcquirePoPTokenInteractive(
+			context.Background(),
+			p.popClaims,
+			scopes,
+			authorityFromConfig.String(),
+			p.clientID,
+		)
+		if err != nil {
+			return emptyToken, fmt.Errorf("failed to create PoP token using interactive login: %w", err)
+		}
+	} else {
+		cred, err := azidentity.NewInteractiveBrowserCredential(&azidentity.InteractiveBrowserCredentialOptions{
+			ClientOptions: clientOpts,
+			TenantID:      p.tenantID,
+			ClientID:      p.clientID,
+		})
+		if err != nil {
+			return emptyToken, fmt.Errorf("unable to create credential. Received: %w", err)
+		}
+
+		// Use the token provider to get a new token
+		interactiveToken, err := cred.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: scopes})
```

Comment:
here you can re-use the `ctx` declared at the top (mentioned on my other comment)

Reply:
Updated to reuse the ctx